File: app/models/unit_vet.rb
# == Schema Information
#
# Table name: unit_vets
#
#  id         :bigint           not null, primary key
#  vet1_desc  :string           not null
#  vet1_exp   :integer          default(0), not null
#  vet2_desc  :string           not null
#  vet2_exp   :integer          default(0), not null
#  vet3_desc  :string           not null
#  vet3_exp   :integer          default(0), not null
#  vet4_desc  :string           not null
#  vet4_exp   :integer          default(0), not null
#  vet5_desc  :string           not null
#  vet5_exp   :integer          default(0), not null
#  created_at :datetime         not null
#  updated_at :datetime         not null
#  unit_id    :bigint           not null
#
# Indexes
#
#  idx_unit_vet_unit_id_uniq   (unit_id) UNIQUE
#  index_unit_vets_on_unit_id  (unit_id)
#
class UnitVet < ApplicationRecord
  belongs_to :unit, inverse_of: :unit_vet

  validates_numericality_of :vet1_exp
  validates_numericality_of :vet2_exp
  validates_numericality_of :vet3_exp
  validates_numericality_of :vet4_exp
  validates_numericality_of :vet5_exp

  validates_presence_of :vet1_desc
  validates_presence_of :vet2_desc
  validates_presence_of :vet3_desc
  validates_presence_of :vet4_desc
  validates_presence_of :vet5_desc
end

File: app/javascript/features/companies/manage/units/unit_vet.js
export const getVetLevel = (exp, unitVet) => {
  if (exp < unitVet.vet1Exp) {
    return [0, unitVet.vet1Exp]
  } else if (exp < unitVet.vet2Exp) {
    return [1, unitVet.vet2Exp]
  } else if (exp < unitVet.vet3Exp) {
    return [2, unitVet.vet3Exp]
  } else if (exp < unitVet.vet4Exp) {
    return [3, unitVet.vet4Exp]
  } else if (exp < unitVet.vet5Exp) {
    return [4, unitVet.vet5Exp]
  } else {
    return [5, null]
  }
}

export const buildVetBonuses = (level, unitVet) => {
  let bonuses = []
  if (level === 0) {
    return bonuses
  }
  if (level >= 1) {
    bonuses.push({ level: 1, desc: unitVet.vet1Desc })
  }
  if (level >= 2) {
    bonuses.push({ level: 2, desc: unitVet.vet2Desc })
  }
  if (level >= 3) {
    bonuses.push({ level: 3, desc: unitVet.vet3Desc })
  }
  if (level >= 4) {
    bonuses.push({ level: 4, desc: unitVet.vet4Desc })
  }
  if (level === 5) {
    bonuses.push({ level: 5, desc: unitVet.vet5Desc })
  }
  return bonuses
}
